### PR TITLE
Eos 28612 : comment and data modal for alert page.

### PR DESCRIPTION
**Pull request by sujit-h-singh**
Signed-off-by: Sujit Singh <sujit.h.singh@seagate.com>

# Problem Statement
- Addition of comment modal and data modal for alert page.

# Design
- Created the structure and component which can be reused as required for alert component

# Coding
   Checklist for Author
-  [x] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [x] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [x] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [x] JIRA number/GitHub Issue added to PR
- [x] PR is self reviewed
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [x] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: src/components/alerts/LrAlertDialog.vue [hunk 1]
```diff
@@ -15,74 +15,76 @@
...
+    </v-dialog>
+  </div>
 </template>
 <script lang="ts">
 import { Component, Vue, Prop, PropSync } from "vue-property-decorator";
 import LrLabel from "../shared/LrLabel/LrLabel.vue";
 @Component({
-  name: "LrAlertDialog",
-  components: {},
+	name: "LrAlertDialog",
```
Comment (NareshGopal) [inline src/components/alerts/LrAlertDialog.vue:44]: Have you changed the tab space configuration for prettier?

File: src/components/alerts/LrAlertInformation.vue [hunk 2]
```diff
@@ -92,117 +92,126 @@
...
       </div>
       <div class="alert-info">
-        <img
-        :src="require(`@/assets/icons/more-info.svg`)"
-        @click="showAlertDetailsDialog = true"
-        class="action-btn-block"
-        alt="logo"
+        <LrSvgIcon
+          icon="more-info.svg"
+          hoverIcon="more-info.svg"
```
Comment (NareshGopal) [inline src/components/alerts/LrAlertInformation.vue:97]: Could you make this prop optional?

File: src/components/shared/LrSvgIcon/LrSvgIcon.vue [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+<script lang="ts">
+import { Component, Vue, Prop } from "vue-property-decorator";
+
+@Component({
+  name: "LrSvgIcon",
+  components: {},
+})
+export default class LrSvgIcon extends Vue {
+  @Prop({ required: true }) private icon: string;
+  @Prop({ required: false }) private hoverIcon: string;
```
Comment (NareshGopal) [inline src/components/shared/LrSvgIcon/LrSvgIcon.vue:44]: The prop is optional but it is required on line 27. If the hoverIcon prop is not passed, use the same icon instead.

Comment (sujit-h-singh) [reply src/components/shared/LrSvgIcon/LrSvgIcon.vue:44]: line no 26  v-if="hoverIcon" so we won't need hover effect if this icon is not provided

File: src/components/alerts/LrAlertInformation.vue (hunk 2 above)
Comment (sujit-h-singh) [reply src/components/alerts/LrAlertInformation.vue:97]: IT is optional

File: src/components/alerts/LrAlertDialog.vue (hunk 1 above)
Comment (sujit-h-singh) [reply src/components/alerts/LrAlertDialog.vue:44]: Tab size is 2 only in my vscode

File: src/components/shared/LrSvgIcon/LrSvgIcon.vue (hunk 3 above)
Comment (NareshGopal) [reply src/components/shared/LrSvgIcon/LrSvgIcon.vue:44]: Okay

**Review by NareshGopal — APPROVED** [review]

File: src/components/alerts/LrAlertInformation.vue (hunk 2 above)
Comment (NareshGopal) [reply src/components/alerts/LrAlertInformation.vue:97]: Then, line 97 could have been removed.

File: src/components/alerts/LrAlertDialog.vue (hunk 1 above)
Comment (NareshGopal) [reply src/components/alerts/LrAlertDialog.vue:44]: Fine. Anyway, we need to have a discussion in setting up the rules for the formatter.

File: src/components/shared/LrSvgIcon/LrSvgIcon.vue (hunk 3 above)
Comment (sujit-h-singh) [reply src/components/shared/LrSvgIcon/LrSvgIcon.vue:45]: change added to handle icon disappear on hover

File: src/components/alerts/LrAlertInformation.vue (hunk 2 above)
Comment (sujit-h-singh) [reply src/components/alerts/LrAlertInformation.vue]: done

File: src/components/shared/LrSvgIcon/LrSvgIcon.vue [hunk 4]
```diff
@@ -15,7 +15,8 @@
 * please email opensource@seagate.com.
 -->
 <template>
-  <div class="icon-container action-btn" v-if="icon">
+  <div class="icon-container" v-if="icon"
+    :class="[hoverIcon ? 'action-btn':'']">
```
Comment (NareshGopal) [inline src/components/shared/LrSvgIcon/LrSvgIcon.vue:19]: Instead of handling with class names, you can change the img src based onMouseOver.

Comment (sujit-h-singh) [reply src/components/shared/LrSvgIcon/LrSvgIcon.vue:19]: css hover will have both the events handled on mouseover , onmouseleave.

Comment (NareshGopal) [reply src/components/shared/LrSvgIcon/LrSvgIcon.vue:19]: There is one more way to handle it using CSS variables in which you can use CSS hover pseudo-class.

Comment (sujit-h-singh) [reply src/components/shared/LrSvgIcon/LrSvgIcon.vue:19]: we will be going with css hover

**Pull request merged**